main.tf

```diff
@@ -39,19 +43,24 @@ module "groups_in_group" {
 }
 
 module "users" {
-  for_each   = var.users
-  depends_on = [module.groups]
-  source     = "./modules/users"
+  for_each = var.users
+  depends_on = [
+    module.groups
+  ]
+  source = "./modules/users"
   providers = {
     googleworkspace = googleworkspace
   }
-  user = each.value
+  user            = each.value
+  random_password = var.random_password
```

Comment:
not sure about this naming. could be named `var.password` as well. in the module it will be used:
`password = var.user.password != null ? var.user.password : var.random_password`